### ISSUE TITLE
[guidance] Virtual Node Backend Configuration not in Envoy Proxy Configuration?

### ISSUE THREAD

**Issue by celestialorb** [aws/aws-app-mesh-controller-for-k8s#415]
**Describe the bug**
I want to start off by stating that I'm new to Envoy and so this could easily be due to my lack of knowledge in its configuration.

I have two services running in EKS, service A and service B, service B talks to service A and I've been trying to get them working in AWS AppMesh with STRICT TLS mode. Service B is able to talk to service A when service A is set to PERMISSIVE TLS mode, but is unable to when it is set to STRICT. The Envoy proxy for Service A simply reports a TLS error: "WRONG_VERSION_NUMBER".

Service B has an AppMesh virtual node with a backend configured for service A. The backend is set to enforce TLS on the port that service A is listening on; however when I dump the configuration of the Envoy proxy for service B, I don't see a reference to this configuration anywhere in it (I believe this would be an "upstream" in Envoy terminology).

Should I expect to see references to the virtual node's backend configuration in the Envoy proxy? For example, should I be able to find a reference to service A's port number in Envoy configuration for service B?

**Steps to reproduce**
Just looking for guidance.

**Expected outcome**
Just looking for guidance.

**Environment**

* App Mesh controller version v1.1.1
* Envoy version v1.15.0.0-prod
* Are you using any integrations? X-ray, Jaeger etc. If so versions? No
* Kubernetes version v1.18
* Using EKS (yes/no), if so version? yes, 1.18

**Additional Context**:

**Comment by achevuru** [comment]
@celestialorb The SSL error that you're seeing usually implies TLS is enabled on one end while disabled on the other end. Do you have TLS enabled under listener for Service A and Backends for Service B? Are you using file based TLS or ACM? Can you share sample VirtualNode specs for both Service A and Service B.

**Comment by celestialorb** [comment]
I eventually figured out my issues with the AppMesh controller, I had two and both stemmed from misconfigurations of AppMesh resources.

My first issue was that my virtual services had no provider specified and thus were not backed by the virtual nodes of the services.

The second issue was that my virtual services names did not match the DNS name my services were using to call those services as I did not realize they were required to match.
